shotover-proxy/examples/cassandra_bench_at_rate.rs

```diff
@@ -0,0 +1,93 @@
+#[path = "../tests/helpers/mod.rs"]
+mod helpers;
+
+use helpers::ShotoverProcess;
+use test_helpers::docker_compose::DockerCompose;
+
+fn main() {
+    let rate = std::env::args().nth(1).expect("Need to provide the message rate as the one and only arg. try values of 1000, 10000, 100000 etc.");
+
+    // The benches and tests automatically set the working directory to CARGO_MANIFEST_DIR.
+    // We need to do the same as the DockerCompose + ShotoverManager types rely on this.
+    std::env::set_current_dir(env!("CARGO_MANIFEST_DIR")).unwrap();
+
+    let latte = Latte::new(rate);
+    {
+        let _compose = DockerCompose::new("examples/cassandra-passthrough/docker-compose.yml")
+            .wait_for_n_t("Startup complete", 1, 90);
+
+        // Uses ShotoverProcess instead of ShotoverManager for a more accurate benchmark
+        let _shotover_manager =
+            ShotoverProcess::new("examples/cassandra-passthrough/topology.yaml");
+
+        println!("Shotover");
+        bench_read(&latte, "localhost:9042");
+        println!("Direct Cassandra");
+        bench_read(&latte, "localhost:9043");
+    }
+
+    println!("Shotover vs Direct Cassandra");
+    latte.compare("read-localhost:9042.json", "read-localhost:9043.json");
+}
+
+fn bench_read(latte: &Latte, address: &str) {
+    latte.bench("read", address)
+}
+
+// TODO: Shelling out directly like this is just for experimenting.
+// Either:
+// * get access to latte as a crate
+// * write our own benchmark logic
+struct Latte {
+    rate: String,
+}
+
+impl Latte {
+    fn new(rate: String) -> Latte {
+        test_helpers::docker_compose::run_command(
+            "cargo",
+            &[
+                "install",
```

Comment:
Can we just add this to our dependencies list and install manually instead of doing this instead the benchmarks?

Reply:
Well we have to install a specific version of latte as it has no stability gaurantees.
And I guess we could document to install this specific version of latte....

But I think in general why list a dependency when we can have things "just work"?

If you are worried about the time it takes to rerun this every time, once its installed it only takes 0.4s to rerun on my machine.
![image](https://user-images.githubusercontent.com/5120858/156099046-4a699201-7ee3-40b6-b4d1-49833bf18a60.png)